### ISSUE TITLE
Enhance notification channel configuration

### ISSUE THREAD

**Issue by MrGeneration** [zammad/zammad#3343]
### Infos:

* Used Zammad version: 3.6
* Installation method (source, package, ..): any selfhosted version
* Operating system: any
* Database + version: any
* Elasticsearch version: any
* Browser + version: any

### Desired behavior:

Zammad provides an extra field "notification sender" *above* the send method selection (SMTP / sendmail). 
This affects:
   * email channel page
   * getting started wizzard
   
This improves usability during setups a lot, because especially on getting started wizard when using "SMTP" the current implementation is very likely to not work and bring confusion. 

Martin wanted the "notification sender" still to be available within email channel settings - at least up to the next major.

Sample hacked version below:
![issue-notification-sender](https://user-images.githubusercontent.com/6549061/102776428-fb920580-438e-11eb-97e6-904e506e2614.gif)

Above shows the getting started wizard.
The notification channel configuration technically looks the same way. At least visually.

### Actual behavior:

When configuring the notification channel without changing the notification sender before, the configuration will most likely fail because most of the times the user won't want to use `noreply@fqdn`. This will be blocked by most mail servers.

### Steps to reproduce the behavior:

* try to configure a notification sender with SMTP that's not allowed to use `noreply@FQDN`
   *  fail because your mail server doesn't like that

**Comment by dominikklein** [comment]
@MrGeneration One question about this one here.

So we would show here during the setup of ah new system the default value `#{config.product_name} <noreply@#{config.fqdn}>`.
And this can then be changed from the admin directly inside the setup. Or would be the exception something different here?

**Comment by MrGeneration** [comment]
>So we would show here during the setup of ah new system the default value #{config.product_name} <noreply@#{config.fqdn}>.
And this can then be changed from the admin directly inside the setup. Or would be the exception something different here?

I think you got that right.
Possibly it's easier to have display name and email address seperated - but I'm not sure here.

**Comment by dominikklein** [comment]
I think to separate it, would be a more complex change. Because its needs then also handled differently in other places. And I think also for hosted it would be a change or?

For me, an additional question would be whether we should show the "plain" (without replacing variables) value or not. I think this question is also the same when it would be split into two fields/configs.

**Comment by dominikklein** [comment]
@MrGeneration and @YetAnotherGerrit maybe you can give some input?

**Comment by YetAnotherGerrit** [comment]
Hosting would also benefit from separating the input fields. Because this way, hosted customers could change the display name of the notification sender without adjusting the (in hosted) fixed e-mail-address.

Can't you separate the input fields, but still join and store that into the same Setting-var?

**Comment by MrGeneration** [comment]
> Hosting would also benefit from separating the input fields. Because this way, hosted customers could change the display name of the notification sender without adjusting the (in hosted) fixed e-mail-address.

Nope. You're mixing notification channel and email accounts. The notification channel cannot be changed in SaaS and that's supposed to stay that way.

> For me, an additional question would be whether we should show the "plain" (without replacing variables) value or not. I think this question is also the same when it would be split into two fields/configs.

I have no hard feelings towards a specific way sorry.

**Comment by MrGeneration** [comment]
This still is an issue and a pitfall for everyone that uses non standard ports for the protocol they use. E.g. `8080`, `8443`.

I thought this was fixed already, but the workflow is still "broken" and a pain.

![Image](https://github.com/user-attachments/assets/77d17583-7538-4459-8b84-bd8420740f9f)

**Comment by dvuckovic** [comment]
We tried to alleviate reported issues with a "quick fix" for #4355, when placeholders in `notification_sender` setting have some offending data. Proper fix would be the proposed split of the notification sender into two values, possibly as part of the notification channel configuration (both in the guided setup + email channel configuration). While defaulting to some system settings does make a lot of sense, I feel like interpolating whatever is set in there is not a good idea from the outset. Perhaps we could switch to a "proposed" value, but still store actual values and rely on them later. This would also allow us to move away from a dedicated setting, since it could be part of the channel configuration.

**Comment by YetAnotherGerrit** [comment]
Related:
https://community.zammad.org/t/change-notification-master-in-commercial-version/17273/2
